src/renderer/html_handlebars/hbs_renderer.rs

```diff
@@ -816,15 +833,19 @@ fn build_header_links(html: &str) -> String {
 fn insert_link_into_header(
     level: usize,
     content: &str,
+    id: Option<String>,
+    classes: Option<String>,
     id_counter: &mut HashMap<String, usize>,
 ) -> String {
-    let id = utils::unique_id_from_content(content, id_counter);
+    let id = id.unwrap_or_else(|| utils::unique_id_from_content(content, id_counter));
+    let classes = classes.unwrap_or("".to_string());
```

Comment:
Can this skip adding `class=""` when there are no classes?

Reply:
Yes, it's possible but it would complicate the code more. Do you think it's really needed to make it have that kind of behavior?

Reply:
Yea, I think it would be better to keep the generated output cleaner. It shouldn't be too complicated, I think something like the following should work:

```rust
let classes = classes.map(|s| format!(" class=\"{s}\"")).unwrap_or_default();
```